### PR TITLE
Thresholds/v20 backoff/v2

### PR DESCRIPTION
**Pull request by victorjulien**
replace #11358 and #11359:
- rebase
- address review comments

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1939

https://redmine.openinfosecfoundation.org/issues/6822
https://redmine.openinfosecfoundation.org/issues/426
https://redmine.openinfosecfoundation.org/issues/6967
https://redmine.openinfosecfoundation.org/issues/7120

### CODE REVIEW

File: src/util-thash.c [hunk 1]
```diff
@@ -407,6 +409,59 @@ int THashWalk(THashTableContext *ctx, THashFormatFunc FormatterFunc, THashOutput
...
+                if (h->prev != NULL)
+                    h->prev->next = h->next;
+                if (h->next != NULL)
+                    h->next->prev = h->prev;
+                if (hb->head == h)
+                    hb->head = h->next;
+                if (hb->tail == h)
+                    hb->tail = h->prev;
+                h->next = NULL;
+                h->prev = NULL;
```
Comment (jasonish) [inline src/util-thash.c:447]: cleanup for later: can this and `THashRemoveFromHash` be refactored to use a common removal routine.

File: src/util-threshold-config.c [hunk 2]
```diff
@@ -1679,15 +1686,15 @@ static int SCThresholdConfTest10(void)
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p1);
     FAIL_IF(PacketTestAction(p1, ACTION_DROP));
     FAIL_IF(PacketAlertCheck(p1, 10) != 1);
-
+#if 0
     /* Ensure that a Threshold entry was installed at the sig */
     FAIL_IF_NULL(de_ctx->ths_ctx.th_entry[s->num]);
-
+#endif
```
Comment (jasonish) [inline src/util-threshold-config.c:1692]: nit: remove or comment why we're disabled

**Review by jasonish — APPROVED** [review]
LGTM now. Still going to run live for a while, and try to test a little more.

**Comment by victorjulien** [comment]
Merged in #11394, thanks!

**Pull request merged**